### ISSUE TITLE
'expression not well-formed' error when parsing PDDL

### ISSUE THREAD

**Issue by MFaisalZaki** [aiplan4eu/unified-planning#454]
Grounder generated 'expression not well-formed'.
To reproduce, 
```
from unified_planning.io import PDDLReader
from unified_planning.shortcuts import *

planning_task = PDDLReader().parse_problem('domain.pddl', 'p1.pddl')
with Compiler(problem_kind=planning_task.kind, compilation_kind=CompilationKind.GROUNDING) as grounder:
     g_task grounder.compile(planning_task, compilation_kind=CompilationKind.GROUNDING).problem
```

I get this error:
```
unified_planning.exceptions.UPTypeError: The expression '(depot0 == market1)' is not well-formed
```
[files.zip](https://github.com/aiplan4eu/unified-planning/files/12049944/files.zip)

**Comment by alvalentini** [comment]
@MFaisalZaki the issue should be fixed by #457. Can you confirm?

**Comment by MFaisalZaki** [comment]
@alvalentini, Yup, it works. Thanks a lot.
